### PR TITLE
docs(jsdoc): Secure Headers Middleware

### PR DESCRIPTION
**Pull request by goisaki**
This PR is to add JSDoc for Secure Headers Middleware.
Note that the target of the PR is not `main`.

Related:
- #1338
- #2680

- [x] `bun denoify` to generate files for Deno
- [x] `bun run format:fix && bun run lint:fix` to format the code

### CODE REVIEW

File: deno_dist/middleware/secure-headers/index.ts [hunk 1]
```diff
@@ -126,6 +126,35 @@ export const NONCE: ContentSecurityPolicyOptionHandler = (ctx) => {
   return `'nonce-${nonce}'`
 }
 
+/**
+ * Secure Headers Middleware for Hono.
+ *
+ * @see {@link https://hono.dev/middleware/builtin/secure-headers}
+ *
+ * @param {Partial<SecureHeadersOptions>} [customOptions] - The options for the secure headers middleware.
```
Comment (yusukebe) [inline deno_dist/middleware/secure-headers/index.ts:134]: Remove `Partial` since #2713 was merged.

Comment (goisaki) [reply deno_dist/middleware/secure-headers/index.ts:134]: Thanks! Resolved and squashed to first commit.

**Comment by goisaki** [comment]
Same problem:

> Hi @yusukebe
> Currently, the change to remove Partial is made only in JSDoc. To include these changes in the code, we need to [merge honojs:main into honojs:docs/jsdoc-middleware](https://github.com/honojs/hono/compare/docs/jsdoc-middleware...main).
> 
> (It is possible to do this merge in this PR, but it is not recommended because it will include out-of-scope changes.)
> 
> _Originally posted by @goisaki in https://github.com/honojs/hono/issues/2703#issuecomment-2119216323_

**Comment by yusukebe** [comment]
Hey @goisaki 

I've updated this PR/branch by myself. It's ready to merge! Go!

**Pull request merged**